core/src/test/java/com/sequenceiq/cloudbreak/service/publicendpoint/FreeIPAEndpointManagementServiceTest.java

```diff
@@ -0,0 +1,191 @@
+import com.sequenceiq.environment.api.v1.environment.model.response.DetailedEnvironmentResponse;
+import com.sequenceiq.freeipa.api.v1.dns.DnsV1Endpoint;
+import com.sequenceiq.freeipa.api.v1.dns.model.AddDnsCnameRecordRequest;
+
+public class FreeIPAEndpointManagementServiceTest {
+
+    private static final String TEST_USER_CRN = "crn:cdp:iam:us-west-1:accid:user:mockuser@cloudera.com";
+
+    private static final String ENVIRONMENT_CRN = "crn:myEnvironment";
+
+    private static final String LB_DNS = "loadbalancer.dns.com";
+
+    private static final String LB_IP = "1.1.1.1";
+
+    private static final String LB_ENDPOINT = "lb-endpoint";
+
+    @Mock
+    private DnsV1Endpoint dnsV1Endpoint;
+
+    @Mock
+    private EnvironmentClientService environmentClientService;
+
+    @Mock
+    private LoadBalancerPersistenceService loadBalancerPersistenceService;
+
+    @Mock
+    private Stack stack;
+
+    @Mock
+    private GrpcUmsClient grpcUmsClient;
+
+    @Mock
+    private EnvironmentBasedDomainNameProvider domainNameProvider;
+
+    @InjectMocks
+    private FreeIPAEndpointManagementService underTest;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        stack = mock(Stack.class);
+        when(stack.getEnvironmentCrn()).thenReturn(ENVIRONMENT_CRN);
+        when(stack.getId()).thenReturn(1L);
+        when(domainNameProvider.getDomainName(any(), any())).thenReturn("cloudera.domain");
+        doNothing().when(dnsV1Endpoint).addDnsCnameRecord(any());
+        doNothing().when(dnsV1Endpoint).addDnsARecord(any());
+        doNothing().when(dnsV1Endpoint).deleteDnsCnameRecord(any(), any(), any());
+        doNothing().when(dnsV1Endpoint).deleteDnsARecord(any(), any(), any());
+
+        DetailedEnvironmentResponse environment = new DetailedEnvironmentResponse();
+        environment.setName("environment-name");
+        when(environmentClientService.getByCrn(any())).thenReturn(environment);
+
+        String accountWorkloadSubdomain = "aWorkloadSubdomain";
+        UserManagementProto.Account umsAccount = UserManagementProto.Account.newBuilder()
+            .setWorkloadSubdomain(accountWorkloadSubdomain)
+            .build();
+        when(grpcUmsClient.getAccountDetails(any(), any(), any())).thenReturn(umsAccount);
+
+        underTest.setCertGenerationEnabled(true);
+    }
+
+    @Test
+    public void testRegisterLoadBalancerWithDNS() {
+        LoadBalancer loadBalancer = new LoadBalancer();
+        loadBalancer.setType(LoadBalancerType.PUBLIC);
+        loadBalancer.setDns(LB_DNS);
+        loadBalancer.setEndpoint(LB_ENDPOINT);
+        when(loadBalancerPersistenceService.findByStackId(any())).thenReturn(Set.of(loadBalancer));
+
+        ThreadBasedUserCrnProvider.doAs(TEST_USER_CRN, () -> underTest.registerLoadBalancerDomainWithFreeIPA(stack));
+
+        ArgumentCaptor<AddDnsCnameRecordRequest> requestCaptor = ArgumentCaptor.forClass(AddDnsCnameRecordRequest.class);
```

Comment:
this really jut a little thing but which one is more correct AddDnsCnameRecordRequest vs AddDns**C**NameRecordRequest 
@lacikaaa

Reply:
CNAME is the original format and I follow the guideline when all the letters are capital, in the variable/class etc only the first one is uppercase. eg:
FreeIPA->FreeIpa
DNS->Dns
IPAddress->IpAddress